libsolutil/Assertions.h

```diff
@@ -40,6 +40,27 @@ namespace solidity::util
 #define ETH_FUNC __func__
 #endif
 
+#if defined(__GNUC__)
+// GCC 4.8+, Clang, Intel and other compilers compatible with GCC (-std=c++0x or above)
+[[noreturn]] inline __attribute__((always_inline)) void unreachable()
+{
+    __builtin_unreachable();
+}
+
+#elif defined(_MSC_VER) // MSVC
+
+[[noreturn]] __forceinline void unreachable()
+{
+    __assume(false);
+}
+
+#else
+
+[[noreturn]] inline void unreachable()
+{
+}
```

Comment:
Did you manage to get this variant with anything? I wonder if the `[[noreturn]]` is enough to make it work.

Reply:
I test-wise disabled the obove #if-branches to force this one. The compiler complains on a `[[noreturn]]` function that still returns. So now I've added a `solThrow(Exception, ...)`. But in most (if not all) supported cases, the two above should hit.